### PR TITLE
Tink build tooling

### PR DESCRIPTION
**Pull request by pokearu**
*Description of changes:*
Build tooling for `tinkerbell/tink` project. The project builds 3 images tink-cli, tink-worker and tink-server.
The tink-server depends on the binaries from `cloudflare/cfssl` project.

The `.patch` is for compatibility support of go1.16 go-licenses, which requires a `+build` comment for all `go:build` files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ AWS_ACCOUNT_ID?=$(shell aws sts get-caller-identity --query Account --output tex
...
 
-PROJECTS?=aws_eks-anywhere brancz_kube-rbac-proxy kubernetes-sigs_cluster-api-provider-vsphere kubernetes-sigs_cri-tools kubernetes-sigs_vsphere-csi-driver jetstack_cert-manager kubernetes_cloud-provider-vsphere plunder-app_kube-vip kubernetes-sigs_etcdadm fluxcd_helm-controller fluxcd_kustomize-controller fluxcd_notification-controller fluxcd_source-controller rancher_local-path-provisioner mrajashree_etcdadm-bootstrap-provider mrajashree_etcdadm-controller tinkerbell_cluster-api-provider-tinkerbell tinkerbell_hegel cloudflare_cfssl
+PROJECTS?=aws_eks-anywhere brancz_kube-rbac-proxy kubernetes-sigs_cluster-api-provider-vsphere kubernetes-sigs_cri-tools kubernetes-sigs_vsphere-csi-driver jetstack_cert-manager kubernetes_cloud-provider-vsphere plunder-app_kube-vip kubernetes-sigs_etcdadm fluxcd_helm-controller fluxcd_kustomize-controller fluxcd_notification-controller fluxcd_source-controller rancher_local-path-provisioner mrajashree_etcdadm-bootstrap-provider mrajashree_etcdadm-controller tinkerbell_cluster-api-provider-tinkerbell tinkerbell_hegel tinkerbell_tink cloudflare_cfssl
```
Comment (abhay-krishna) [inline Makefile:6]: let's put this later since this consumes the cfssl one.

File: projects/tinkerbell/tink/docker/linux/tink-worker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+ARG BASE_IMAGE # https://gallery.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base
+FROM $BASE_IMAGE
+
+ARG TARGETARCH
+ARG TARGETOS
+
+COPY _output/bin/tink/$TARGETOS-$TARGETARCH/tink-worker /usr/bin/tink
```
Comment (abhay-krishna) [inline projects/tinkerbell/tink/docker/linux/tink-worker/Dockerfile:7]:
```suggestion
COPY _output/bin/tink/$TARGETOS-$TARGETARCH/tink-worker /usr/bin/tink-worker
```

File: projects/tinkerbell/tink/Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
+GIT_TAG?=$(shell cat GIT_TAG)
+GOLANG_VERSION?="1.17"
+
+REPO=tink
+REPO_OWNER=tinkerbell
+
+LICENSE_PACKAGE_FILTER?=./cmd/tink-cli ./cmd/tink-server ./cmd/tink-worker
```
Comment (abhay-krishna) [inline projects/tinkerbell/tink/Makefile:8]: i think the SOURCE_PATTERNS will cover this

File: projects/tinkerbell/tink/Makefile [hunk 4]
```diff
@@ -0,0 +1,31 @@
+BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
```
Comment (jaxesn) [inline projects/tinkerbell/tink/Makefile:1]: Pretty clean makefile!

Comment (pokearu) [reply projects/tinkerbell/tink/Makefile:1]: Thank you 😄

File: projects/tinkerbell/tink/patches/0001-Adding-build-comment-for-go1.16-go-licenses-compatab.patch [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ tools.go | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/tools.go b/tools.go
+index 177df05..5091dcb 100644
+--- a/tools.go
++++ b/tools.go
+@@ -1,4 +1,5 @@
+ //go:build tools
++// +build
```
Comment (abhay-krishna) [inline projects/tinkerbell/tink/patches/0001-Adding-build-comment-for-go1.16-go-licenses-compatab.patch:16]: This should be `// +build tools`

**Comment by pokearu** [comment]
/override bottlerocket-bootstrap-tooling-presubmit
/override cert-manager-tooling-presubmit
/override cloud-provider-vsphere-tooling-presubmit
/override cluster-api-provider-aws-tooling-presubmit
/override cluster-api-provider-tinkerbell-tooling-presubmit
/override cluster-api-provider-vsphere-tooling-presubmit
/override cluster-api-tooling-presubmit
/override cri-tools-tooling-presubmit
/override eks-anywhere-cli-tools-tooling-presubmit
/override eks-anywhere-diagnostic-collector-tooling-presubmit
/override eks-anywhere-test-image-tooling-presubmit
/override etcdadm-bootstrap-provider-tooling-presubmit
/override etcdadm-controller-tooling-presubmit
/override etcdadm-tooling-presubmit
/override flux-tooling-presubmit
/override govmomi-tooling-presubmit
/override hegel-tooling-presubmit
/override helm-controller-tooling-presubmit
/override imagebuilder-presubmit
/override kind-tooling-presubmit
/override kube-rbac-proxy-tooling-presubmit
/override kube-vip-tooling-presubmit
/override kustomize-controller-tooling-presubmit
/override local-path-provisioner-tooling-presubmit
/override notification-controller-tooling-presubmit
/override source-controller-tooling-presubmit
/override troubleshoot-tooling-presubmit
/override vsphere-csi-driver-tooling-presubmit
/override cfssl-tooling-presubmit

**Comment by pokearu** [comment]
/override bottlerocket-bootstrap-tooling-presubmit
/override cert-manager-tooling-presubmit
/override cloud-provider-vsphere-tooling-presubmit
/override cluster-api-provider-aws-tooling-presubmit
/override cluster-api-provider-tinkerbell-tooling-presubmit
/override cluster-api-provider-vsphere-tooling-presubmit
/override cluster-api-tooling-presubmit
/override cri-tools-tooling-presubmit
/override eks-anywhere-cli-tools-tooling-presubmit
/override eks-anywhere-diagnostic-collector-tooling-presubmit
/override eks-anywhere-test-image-tooling-presubmit
/override etcdadm-bootstrap-provider-tooling-presubmit
/override etcdadm-controller-tooling-presubmit
/override etcdadm-tooling-presubmit
/override flux-tooling-presubmit
/override govmomi-tooling-presubmit
/override hegel-tooling-presubmit
/override helm-controller-tooling-presubmit
/override imagebuilder-presubmit
/override kind-tooling-presubmit
/override kube-rbac-proxy-tooling-presubmit
/override kube-vip-tooling-presubmit
/override kustomize-controller-tooling-presubmit
/override local-path-provisioner-tooling-presubmit
/override notification-controller-tooling-presubmit
/override source-controller-tooling-presubmit
/override troubleshoot-tooling-presubmit
/override vsphere-csi-driver-tooling-presubmit
/override cfssl-tooling-presubmit

**Pull request merged**